repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/core/classloader/ClassLoaderTest.groovy

```diff
@@ -104,7 +95,6 @@ class ClassLoaderTest extends ReposeValveTest {
         then: "the request header should equal BAR"
         mc.handlings.size() == 1
         mc.receivedResponse.code == "200"
-        mc.handlings[0].request.headers.getFirstValue("FOO") == "BAR"
```

Comment:
So the header isn't actually being replaced, a ethod is being overwriten to make it look like it's replaced. There is no check we can of the deproxy objects that will fire off that method. Thankfully the check does still happen, but it happens in the filter after the one with the method. This is also why as as soon as i turned on intrafilter logging things started breaking. Intrafilter logging applies our own wrappers between filters, so i was losing the method call.